cider-dap/src/adapter.rs

```diff
@@ -199,26 +218,36 @@ impl MyAdapter {
             Some(p) => {
                 let out: Vec<Variable> = p
                     .iter()
-                    .map(|x| Variable {
-                        name: String::from(x),
-                        value: String::from("1"),
-                        type_field: None,
-                        presentation_hint: None,
-                        evaluate_name: None,
-                        variables_reference: 0,
-                        named_variables: None,
-                        indexed_variables: None,
-                        memory_reference: None,
+                    .map(|(nam, val)| {
+                        let valu = match val.as_option() {
+                            None => 0,
+                            Some(assigned_val) => {
+                                assigned_val.val().to_u64().unwrap()
+                            }
+                        };
+                        Variable {
+                            name: String::from(nam),
+                            value: valu.to_string(),
+                            type_field: None,
+                            presentation_hint: None,
+                            evaluate_name: None,
+                            variables_reference: 0,
+                            named_variables: None,
+                            indexed_variables: None,
+                            memory_reference: None,
+                        }
                     })
                     .collect();
                 out
             }
         }
     }
-    // return cells in calyx context (later should hopefully just return ones w current component)
-    pub fn get_scopes(&mut self, _frame: i64) -> Vec<Scope> {
+    // return cells in calyx context
+    // todo: return only cells in current stack frame (component)
+    pub fn get_scopes(&mut self, frame: i64) -> Vec<Scope> {
         let mut out_vec = vec![];
-        let cell_names = self.debugger.get_cells();
+        let component = self.frames_to_cmpts.get(&frame).unwrap();
```

Comment:
If you're going to use `get(...).unwrap()` might as well use direct indexing
```suggestion
        let component = self.frames_to_cmpts[&frame];
```